src/ai/chatAi.ts

```diff
@@ -46,6 +52,62 @@ export default class ChatAi {
     return this.ain.sendTransaction(txBody);
   }
 
+  async depositCredit(
+    provider: string,
+    api: string,
+    amount: number
+  ): Promise<CreditDepositTransactionResult> {
+    const address = this.ain.signer.getAddress();
+
+    const aiName = validateAndGetAiName(provider, api);
+    const aiService = await validateAndGetAiService(aiName, this.ainize);
+
+    // TODO(jiyoung): update login method to use signer. (if implemented)
+    await AinizeAuth.getInstance().login(this.ain, this.ainize);
+    const currentBalance = await aiService.getCreditBalance();
+    const txHash = await aiService.chargeCredit(amount);
+    const updatedBalance = await this.waitForCreditUpdate(
+      currentBalance + amount,
+      60 * 1000, // 1 min
+      aiService
+    );
+    await AinizeAuth.getInstance().logout(this.ainize);
+
+    return { tx_hash: txHash, address, balance: updatedBalance };
+  }
+
+  withdrawCredit(): never {
+    throw new Error('Not implemented');
+  }
+
+  async getCredit(provider: string, api: string): Promise<number> {
+    const aiName = validateAndGetAiName(provider, api);
+    const aiService = await validateAndGetAiService(aiName, this.ainize);
+
+    // TODO(jiyoung): update login method to use signer. (if implemented)
+    await AinizeAuth.getInstance().login(this.ain, this.ainize);
+    const balance = await aiService.getCreditBalance();
+    await AinizeAuth.getInstance().logout(this.ainize);
+
+    return balance;
+  }
+
+  private async waitForCreditUpdate(
+    expectedBalance: number,
+    timeout: number,
+    service: Service
+  ) {
+    const startTime = Date.now();
+    while (Date.now() - startTime < timeout) {
+      const balance = await service.getCreditBalance();
+      if (balance === expectedBalance) {
+        return balance;
+      }
+      await new Promise((resolve) => setTimeout(resolve, 5000)); // 5 sec
```

Comment:
(minor) maybe this can be added as a util function. ref: https://github.com/ainblockchain/ain-blockchain/blob/master/common/common-util.js#L896

Reply:
Done!